apis/v1beta1/collector_webhook_test.go

```diff
@@ -158,9 +159,19 @@ func TestCollectorDefaultingWebhook(t *testing.T) {
 					Mode:            v1beta1.ModeDeployment,
 					UpgradeStrategy: v1beta1.UpgradeStrategyAutomatic,
 					Config: func() v1beta1.Config {
-						const input = `{"receivers":{"otlp":{"protocols":{"grpc":{"endpoint":"0.0.0.0:4317"},"http":{"endpoint":"0.0.0.0:4318"}}}},"exporters":{"debug":null},"service":{"telemetry":{"metrics":{"address":"0.0.0.0:8888"}},"pipelines":{"traces":{"receivers":["otlp"],"exporters":["debug"]}}}}`
+						const input = `{"receivers":{"otlp":{"protocols":{"grpc":{"endpoint":"0.0.0.0:4317"},"http":{"endpoint":"0.0.0.0:4318"}}}},"exporters":{"debug":null},"service":{"telemetry":{"metrics":{"readers":[{"pull":{"exporter":{"prometheus":{"host":"0.0.0.0","port":8888}}}}]}},"pipelines":{"traces":{"receivers":["otlp"],"exporters":["debug"]}}}}`
 						var cfg v1beta1.Config
 						require.NoError(t, yaml.Unmarshal([]byte(input), &cfg))
+
+						r := cfg.Service.Telemetry.Object["metrics"].(map[string]interface{})["readers"]
+						readers := r.([]interface{})
+						firstReader := readers[0].(map[string]interface{})
+						exporter := firstReader["pull"].(map[string]interface{})["exporter"].(map[string]interface{})
+						// Convert port to float64 to avoid type mismatch because how go-yaml unmarshals
+						exporter["prometheus"].(map[string]interface{})["port"] = float64(8888)
+
+						// This is a workaround to avoid the type mismatch
+						exporter["AdditionalProperties"] = nil
```

Comment:
Can you create an issue to migrate to https://github.com/goccy/go-yaml and fix these by using [custom unmarshalers](https://pkg.go.dev/github.com/goccy/go-yaml#CustomUnmarshaler)?

Reply:
I think I will need to implement this right now. The thing is the marshaling in the collector is producing errors because of this additional field...

Using the `config` package from the collector is producing some headaches...

Reply:
That's the thing.. it didn't and is producing error when the config is provided to the collector. I'm looking for a good way to do it.